cloud/scope/machine.go

```diff
@@ -334,6 +334,10 @@ func (m *MachineScope) DeleteVPCLoadBalancerPoolMember() error {
 		return nil
 	}
 
+	if m.IBMVPCMachine.Status.InstanceID == "" {
```

Comment:
How about moving this check to begining of this function itself, so we can avoid GetLoadbaalncer() call also?

Reply:
makes sense, moved it, PTAL